Userland/Libraries/LibCrypto/Authentication/GHash.h

```diff
@@ -35,8 +35,12 @@ class GHash final {
 
     explicit GHash(const ReadonlyBytes& key)
     {
-        for (size_t i = 0; i < 16; i += 4)
-            m_key[i / 4] = AK::convert_between_host_and_big_endian(*(const u32*)(key.offset(i)));
+        VERIFY(key.size() >= 16);
+        for (size_t i = 0; i < 16; i += 4) {
+            u32 key_host;
+            __builtin_memcpy(&key_host, key.offset(i), 4);
```

Comment:
You might be interested to know that `AK/BinaryReader.h` (I think that's what I named it...?) is a thing that does this in _one_ line!
Exciting!
Please use that instead 🙃

Reply:
That's neat. a feels like it's both verbose and obfuscated what's going on, but at least it's one line :)

Reply:
Improvements are welcome, I threw that together because I got bored of typing `__builtin_memcpy` when I was fixing misaligned reads in LibTLS :P